From a2c763a16916dd154216174d9bf459c743f5e5d9 Mon Sep 17 00:00:00 2001
From: Fushan Wen <qydwhotmail@gmail.com>
Date: Wed, 21 Feb 2024 10:56:55 +0800
Subject: [PATCH] applets/taskmanager: simplify right press handling

Unfortunately, Taphandler's pressed event only triggers when the press
is lifted for RMB on Wayland, so we need to use the longPressed signal
since it triggers when the button is first pressed.


(cherry picked from commit f2160c3b68c4adcac6c399e14a88b80c23343d55)
---
 applets/taskmanager/package/contents/ui/Task.qml | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/applets/taskmanager/package/contents/ui/Task.qml b/applets/taskmanager/package/contents/ui/Task.qml
index 2852abc5a0..bc8b011419 100644
--- a/applets/taskmanager/package/contents/ui/Task.qml
+++ b/applets/taskmanager/package/contents/ui/Task.qml
@@ -189,7 +189,7 @@ PlasmaCore.ToolTipArea {
     }
     onAudioIndicatorsEnabledChanged: task.hasAudioStreamChanged()
 
-    Keys.onMenuPressed: contextMenuTimer.start()
+    Keys.onMenuPressed: menuTapHandler.longPressed()
     Keys.onReturnPressed: TaskTools.activateTask(modelIndex(), model, event.modifiers, task, Plasmoid, tasks)
     Keys.onEnterPressed: Keys.returnPressed(event);
     Keys.onSpacePressed: Keys.returnPressed(event);
@@ -317,15 +317,10 @@ PlasmaCore.ToolTipArea {
 
     TapHandler {
         acceptedButtons: Qt.RightButton
-        acceptedDevices: PointerDevice.Mouse | PointerDevice.TouchPad
+        acceptedDevices: PointerDevice.Mouse | PointerDevice.TouchPad | PointerDevice.Stylus
         gesturePolicy: TapHandler.WithinBounds // Release grab when menu appears
-        onPressedChanged: if (pressed) contextMenuTimer.start()
-    }
-
-    Timer {
-        id: contextMenuTimer
-        interval: 0
-        onTriggered: menuTapHandler.longPressed()
+        longPressThreshold: 0.001 // https://invent.kde.org/qt/qt/qtdeclarative/-/commit/8f6809681ec82da783ae8dcd76fa2c209b28fde6
+        onLongPressed: menuTapHandler.longPressed()
     }
 
     TapHandler {
-- 
GitLab

